topics/dev/tutorials/interactive-tools/tutorial.md

```diff
@@ -836,10 +817,39 @@ In the case of our `Tabulator` application, the run script is simply the R scrip
 ## Templated config files
 Using the `<configfiles>` section in the tool XML, we can enable complex user configuration for the applcation by templating a run script or configuration file to be read by the application. In this application for example, we could use a `<configfiles>` section to template user input into the `app.R` script that runs the application within the Docker container. This could enable the user to customize the layout of the app before launch.
 
+## Galaxy history interaction
+We have demonstrated how to pass an input file to the Docker container. But what if the application needs to interact with the user's Galaxy history? For example, if the user creates a file within the application. That's where the environment variables created in the tool XML become useful.
+
+> ### {% icon tip %} Access histories in R
+> From the [R-Studio]() GxIT we can see that there is an R library that allows us to communicate with Galaxy histories:
+>
+> "The convenience functions `gx_put()` and `gx_get()` are available to you to interact with your current Galaxy history. You can save your workspace with `gx_save()`"
+>
+>
+{: .tip}
+
```

Comment:
Here we can mention that:
- to do this you need to use Galaxy's API (directly or using libraries like bioblend)
- you can find example implementations in https://github.com/hexylena/rGalaxyConnector which itself uses https://github.com/bgruening/galaxy_ie_helpers